File: manager/python/lib/clear_directus_cache.py
import os
from urllib import request

from utils import logger


def clear_directus_cache():
    url = (
        os.environ.get("DIRECTUS_PUBLIC_URL", "http://directus:8055")
        + "/utils/cache/clear?access_token="
        + os.environ.get("ADMIN_TOKEN", "")
    )
    try:
        request.urlopen(request.Request(url, method="POST"))
        logger.info("Directus cache cleared successfully")
    except Exception as e:
        logger.info(f"Directus cache clearing failed: {e}")

File: manager/javascript/tasks/clearDirectusCache.mjs
export default async function (_payload, helpers) {
  const { logger } = helpers;
  logger.info("Received clearDirectusCache task");
  const directusUrl = process.env.DIRECTUS_PUBLIC_URL || "http://directus:8055";
  try {
    const res = await fetch(directusUrl + "/utils/cache/clear", {
      method: "POST",
      headers: { Authorization: `Bearer ${process.env.ADMIN_TOKEN}` },
    });
    logger.info("Clear Directus cache response status: " + res.status);
  } catch (error) {
    logger.error(error);
  }
}
